crates/uv-interpreter/src/python_version.rs

```diff
@@ -31,6 +31,12 @@ impl FromStr for PythonVersion {
         if version.epoch() != 0 {
             return Err(format!("Python version {s} has a non-zero epoch"));
         }
+        if version.is_post() {
+            return Err(format!("Python version {s} is a post-release"));
+        }
+        if version.is_pre() {
+            return Err(format!("Python version {s} is a pre-release"));
+        }
```

Comment:
I guess we can support this, but it makes the code below simpler because we can construct `markers.python_full_version` by _just_ looking at the releases. Otherwise, I need to add some more handling to it.